File: src/main/java/org/example/controller/LoginController.java
package org.example.controller;

import jakarta.servlet.http.HttpSession;
import org.example.dto.CreateUserRequest;
import org.example.dto.UserResponse;
import org.example.service.UserService;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class LoginController {

    private final UserService userSvc; // 약어

    public LoginController(UserService userSvc) {
        this.userSvc = userSvc;
    }

    @GetMapping("/signup")
    public String signupForm() {
        return "signup";
    }

    @PostMapping("/signup")
    public String signup(@RequestParam String username,
                         @RequestParam String password,
                         @RequestParam String nickname) {
        // 회원가입 처리
        CreateUserRequest req = new CreateUserRequest(username, password, nickname);
        userSvc.create(req);
        return "redirect:/login";
    }

    @PostMapping("/login")
    public String login(@RequestParam String username,
                        @RequestParam String password,
                        HttpSession session) {
        UserResponse user = userSvc.authenticate(username, password);
        if (user != null) {
            session.setAttribute("user", user);
            return "redirect:/";
        }
        // 로그인 실패
        return "redirect:/login?error";
    }

    @PostMapping("/logout")
    public String logout(HttpSession session) {
        session.invalidate();
        return "redirect:/login";
    }
}

File: src/main/kotlin/org/example/controller/LoginController.kt
package org.example.controller

import jakarta.servlet.http.HttpSession
import org.example.dto.CreateUserRequest
import org.example.service.UserService
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestParam

@Controller
class LoginController(
    private val userService: UserService
) {

    @GetMapping("/signup")
    fun signupPage(): String {
        return "signup"
    }

    @PostMapping("/signup")
    fun signup(
        @RequestParam username: String,
        @RequestParam password: String,
        @RequestParam nickname: String
    ): String {
        val request = CreateUserRequest(
            username = username,
            password = password,
            nickname = nickname
        )
        userService.create(request)
        return "redirect:/login"
    }

    @PostMapping("/login")
    fun login(
        @RequestParam username: String,
        @RequestParam password: String,
        session: HttpSession
    ): String {
        val user = userService.authenticate(username, password)
        if (user != null) {
            session.setAttribute("user", user)
            return "redirect:/"
        }
        return "redirect:/login?error"
    }

    @PostMapping("/logout")
    fun logout(session: HttpSession): String {
        session.invalidate()
        return "redirect:/login"
    }
}
